src/main/kotlin/it/pagopa/wallet/services/WalletService.kt

```diff
@@ -64,25 +63,41 @@ class WalletService(
         serviceList: List<ServiceName>,
         userId: UUID,
         paymentMethodId: UUID
-    ): Mono<LoggedAction<Wallet>> {
+    ): Mono<Pair<LoggedAction<Wallet>, URI>> {
         logger.info("Create wallet with payment methodId: $paymentMethodId and userId: $userId")
         return ecommercePaymentMethodsClient
             .getPaymentMethodById(paymentMethodId.toString())
             .map {
-                return@map Wallet(
-                    id = WalletId(UUID.randomUUID()),
-                    userId = UserId(userId),
-                    status = WalletStatusDto.CREATED,
-                    paymentMethodId = PaymentMethodId(UUID.fromString(it.id)),
-                    version = 0,
-                    creationDate = Instant.now(),
-                    updateDate = Instant.now()
+                val creationTime = Instant.now()
+                return@map Pair(
+                    Wallet(
+                        id = WalletId(UUID.randomUUID()),
+                        userId = UserId(userId),
+                        status = WalletStatusDto.CREATED,
+                        paymentMethodId = PaymentMethodId(paymentMethodId),
+                        version = 0,
+                        creationDate = creationTime,
+                        updateDate = creationTime
+                    ),
+                    it
                 )
             }
-            .flatMap { wallet ->
-                walletRepository.save(wallet.toDocument()).map {
-                    LoggedAction(it.toDomain(), WalletAddedEvent(it.id))
-                }
+            .flatMap { (wallet, paymentMethodResponse) ->
+                walletRepository
+                    .save(wallet.toDocument())
+                    .map { LoggedAction(wallet, WalletAddedEvent(wallet.id.value.toString())) }
+                    .map { loggedAction ->
+                        Pair(
+                            loggedAction,
+                            paymentMethodResponse.name.let {
+                                when (WalletDetailsType.valueOf(it)) {
+                                    WalletDetailsType.CARDS ->
+                                        onboardingReturnUrlConfig.cardReturnUrl
+                                    else -> onboardingReturnUrlConfig.apmReturnUrl
```

Comment:
I would manage explicitly the `WalletDetailsType.APM` and let and else with an empty value or a managed exception. This is not properly for this pr, but for future develop, if we add another details type we could fall in this else. 
What do you think?

Reply:
Yes, I like it, but I would rather raise a managed exception and map it to a 400 bad request :D working on it ;)

Reply:
Make else case explicit so that if a new enum constant is add in `WalletDetailsType` a compilation error is raised.
There is no need to explicit map to 400 bad request here since this check is already performed into [EcommercePaymentMethodsClient](https://github.com/pagopa/pagopa-wallet-service/blob/bd63e2d2d497d2699525bdd4f69f0ccf92c53876/src/main/kotlin/it/pagopa/wallet/client/EcommercePaymentMethodsClient.kt#L80-L82) where, if an enabled payment method that has not been mapped to this enumeration is returned by payment methods raises a 400 bad request during wallet creation